towhee/base_repr.py

```diff
@@ -25,30 +22,32 @@ class BaseRepr:
         name:
             Name of the internal object described by this representation.
     """
-
-    def __init__(self, name: str):
-        self._name = name
+    def __init__(self, name: str = None):
+        if name:
+            self._name = name
 
     @property
     def name(self):
         return self._name
 
-    def serialize(self) -> str:
-        """Universal function used to serialize this representation.
-
-        Returns:
-            A string containing the serialized version of this representation. Example
-            output:
-
-            'VariableRepr:
-                name: variable
-                vtype: tensor
-                dtype: float'
-        """
-        out = self.__class__.__name___ + '\n'
-        for name in dir(self):
-            value = self.__getattribute__(name)
-            if name[0] != '_' and not isinstance(value, types.FunctionType):
-                value = value.__class__.__name__
-                out += name + ': ' + str(value) + '\n'
-        return out
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+    # def serialize(self) -> str:
```

Comment:
This implementation of `serialize` is no longer correct based on our latest arch, remove this commented block and add a `raise NotImplementedError` instead.

Reply:
fixed